tests/tracer/test_ext.py

```diff
@@ -23,5 +26,118 @@ def _updateenv(monkeypatch, env):
+        assert mock_popen.call_args[0][0] == expected_call_format
+        extracted_tags = ci.tags()
+
+    assert extracted_committer == expected_committer
+    assert extracted_tags["git.commit.committer.name"] == expected_committer[0]
+    assert extracted_tags["git.commit.committer.email"] == expected_committer[1]
+    assert extracted_tags["git.commit.committer.date"] == expected_committer[2]
+
+
+def test_git_extract_user_info_error():
+    """On error, the author/committer tags should not be extracted, and should internally raise and log the error."""
+    with pytest.raises(ValueError):
+        with mock.patch("ddtrace.ext.ci.git.subprocess.Popen") as mock_popen:
+            mock_popen.return_value.returncode = -1
+            mock_popen.return_value.communicate.return_value = (b"", b"")
+            git.extract_user_info(author=True)
+
+
+def test_git_extract_repository_url():
+    """Make sure that the git repository url is extracted properly."""
+    expected_repository_url = "https://github.com/scope-demo/scopeagent-reference-springboot2.git"
+    mock_repository_url_output = b"https://github.com/scope-demo/scopeagent-reference-springboot2.git"
+    with mock.patch("ddtrace.ext.ci.git.subprocess.Popen") as mock_popen:
+        mock_popen.return_value.returncode = 0
+        mock_popen.return_value.communicate.return_value = (mock_repository_url_output, b"")
+        extracted_repository_url = git.extract_repository_url()
+
+    assert extracted_repository_url == expected_repository_url
+
+
+def test_git_extract_repository_url_error():
+    """On error, the repository url tag should not be extracted, and should internally raise and log the error."""
+    with pytest.raises(ValueError):
+        with mock.patch("ddtrace.ext.ci.git.subprocess.Popen") as mock_popen:
+            mock_popen.return_value.returncode = -1
+            mock_popen.return_value.communicate.return_value = (b"", b"")
+            git.extract_repository_url()
+
+
+def test_git_extract_commit_message():
+    """Make sure that the git commit message is extracted properly."""
+    expected_msg = "Update README.md"
+    mock_output = b"Update README.md"
+    with mock.patch("ddtrace.ext.ci.git.subprocess.Popen") as mock_popen:
+        mock_popen.return_value.returncode = 0
+        mock_popen.return_value.communicate.return_value = (mock_output, b"")
+        extracted_msg = git.extract_commit_message()
+
+    assert extracted_msg == expected_msg
+
+
+def test_git_extract_commit_message_error():
+    """On error, the commit message tag should not be extracted, and should internally raise and log the error."""
+    with pytest.raises(ValueError):
+        with mock.patch("ddtrace.ext.ci.git.subprocess.Popen") as mock_popen:
+            mock_popen.return_value.returncode = -1
+            mock_popen.return_value.communicate.return_value = (b"", b"")
+            git.extract_commit_message()
+
+
+def test_git_executable_not_found_error():
```

Comment:
do we have any tests for `extract_git_metadata`?

Reply:
I can add one in 😄